src/main/webapp/dev/js/main/feedbackSubmissionsEdit.es6

```diff
@@ -888,10 +888,19 @@ function updateRankMessageQn(qnNum) {
     }
 
     function checkMinMaxRestrictions(questionNumber, recipientIndex) {
-        const rankedOptions = $(`select[name="responsetext-${questionNumber}-${recipientIndex}"]`)
+        let rankedOptions = 0;
+
+        if ($(`#rankToRecipients-${questionNumber}`).val() === 'true') {
+            rankedOptions = $(`select[name^="responsetext-${questionNumber}-"]`)
+                              .filter(function () {
+                                  return $(this).val() !== '';
+                              }).length;
+        } else {
+            rankedOptions = $(`select[name="responsetext-${questionNumber}-${recipientIndex}"]`)
```

Comment:
If the difference is only in the selector name, consider confining the `if` condition to just that.
```js
const selector = $(`#rankToRecipients-${questionNumber}`).val() === 'true' ? ... : ...;
const rankedOptions = $(...).filter(...).length;
```